test/Microsoft.AspNet.Mvc.Core.Test/ControllerTests.cs

```diff
@@ -374,6 +374,139 @@ public void RedirectToRoutePermanent_WithParameterRouteName_SetsResultSameRouteN
         }
 
         [Fact]
+        public void Created_WithStringParameter_SetsCreatedLocation()
+        {
+            // Arrange
+            var controller = new Controller();
+            var uri = "http://test/url";
+
+            // Act
+            var result = controller.Created(uri, null);
+
+            // Assert
+            Assert.IsType<CreatedResult>(result);
+            Assert.Same(uri, result.Location);
+        }
+
+        [Fact]
+        public void Created_WithUriParameter_SetsCreatedLocation()
+        {
+            // Arrange
+            var controller = new Controller();
+            var uri = new Uri("http://test/url");
+
+            // Act
+            var result = controller.Created(uri, null);
+
+            // Assert
+            Assert.IsType<CreatedResult>(result);
+            Assert.Same(uri.AbsoluteUri, result.Location);
+        }
+
+        [Fact]
+        public void CreatedAtAction_WithParameterActionName_SetsResultActionName()
+        {
+            // Arrange
+            var controller = new Controller();
+
+            // Act
+            var result = controller.CreatedAtAction("SampleAction", null);
+
+            // Assert
+            Assert.IsType<CreatedAtActionResult>(result);
+            Assert.Equal("SampleAction", result.ActionName);
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData(null)]
+        [InlineData("SampleController")]
+        public void CreatedAtAction_WithActionControllerAndNullRouteValue_SetsSameValue(
+            string controllerName)
+        {
+            // Arrange
+            var controller = new Controller();
+
+            // Act
+            var result = controller.CreatedAtAction("SampleAction", controllerName, null, null);
+
+            // Assert
+            Assert.IsType<CreatedAtActionResult>(result);
+            Assert.Equal("SampleAction", result.ActionName);
+            Assert.Equal(controllerName, result.ControllerName);
+        }
+
+        [Theory]
+        [MemberData(nameof(CreatedAndRedirectTestData))]
```

Comment:
I don't see the purpose of running this with multiple values for `routeData`. What *different* behaviors are you trying to verify?